### PR TITLE
Chamber dash

### PR DESCRIPTION
**Pull request by Blamcho**
It was implemented in the lava level in the scene along with lava balls that go up and down between platforms, horizontal lava cannons were also added and in the part of the level where the lava rises, a simple version of lava that rises was added.

https://github.com/user-attachments/assets/278a9722-3b49-4c11-8e32-d37d1a1e992e

### CODE REVIEW

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaBallMovement.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+using UnityEngine;
+
+public class LavaBallMovement : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaBallMovement.cs:3]: All clases need a namespace.

Comment (Horothenic) [reply Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaBallMovement.cs:3]: This needs a destroy state, when is going to be destroyed?

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectil.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+using UnityEngine;
+
+public class LavaProyectil : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectil.cs:3]: Rename as `LavaProjectile`

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectil.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public float speed = 5f;          
+    public float tiempoDeVida = 8f;
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectil.cs:6]: Use english and not publc fields, change to `[SerializeField] private` in all fields in all scripts

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectil.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+public class LavaProyectil : MonoBehaviour
+{
+    public float speed = 5f;          
+    public float tiempoDeVida = 8f;   
+    private Rigidbody2D rb;
+
+    void Start()
+    {
+
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectil.cs:11]: Remove this whitespace.

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectilLeft.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+using UnityEngine;
+
+public class LavaProyectilLeft : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectilLeft.cs:3]: Remove this script and reuse the other, you can make things go to left using negative velocity.

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaRise.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LavaRise : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaRise.cs:5]: I remember chambers where the lava needs to stop, where is that code?

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavacanonLeft.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
+using UnityEngine;
+
+public class LavacanonLeft : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavacanonLeft.cs:3]: Rename as `LavaCannon`

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavacanonRight.cs [hunk 8]
```diff
@@ -0,0 +1,32 @@
+using UnityEngine;
+
+public class LavacanonRight : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavacanonRight.cs:3]: Refactor the other one so only one cannon script exist, add a variable to specify the direction of shooting.

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/PLatformFails.cs [hunk 9]
```diff
@@ -0,0 +1,25 @@
+using System.Collections;
+using UnityEngine;
+
+public class PLatformFails : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/PLatformFails.cs:4]: Rename `CrumblingPlatform`

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavacanonLeft.cs [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    void Update()
+    {
+        if (Time.time >= nextFireTime)
+        {
+            Disparar();
+            nextFireTime = Time.time + Random.Range(minFireRate, maxFireRate);
+        }
+    }
+
+    void Disparar()
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavacanonLeft.cs:26]: Use english please!!!

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavacanonLeft.cs [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+{
+    public GameObject lavaProyectilPrefab; 
+    public Transform firePoint;            
+    public float minFireRate = 2f;         
+    public float maxFireRate = 5f;        
+    private float nextFireTime = 0f;
+
+    void Start()
+    {
+
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavacanonLeft.cs:13]: Remove whitespace.

**Review by Horothenic — CHANGES_REQUESTED** [review]

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectilLeft.cs (hunk 5 above)
Comment (Blamcho) [reply Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaProyectilLeft.cs:3]: already added it, it's called lavaTrigger

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaRise.cs (hunk 6 above)
Comment (Blamcho) [reply Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaRise.cs:5]: already added it, it's called lavaTrigger

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaRise.cs [hunk 12]
```diff
@@ -1,30 +1,45 @@
...
+            PlayerManager.Instance.Respawn.RespawnAction -= resetLava;
         }
-    }
 
-    public void ResetLava()
-    {
-        transform.position = initialPosition;
+        void Update()
+        {
+            Debug.Log($"_isRising: {_isRising}, PositionY: {transform.position.y}");
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaRise.cs:26]: Is this still needed?

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaRise.cs [hunk 13]
```diff
@@ -1,30 +1,45 @@
...
+                transform.position += Vector3.up * _riseSpeed * Time.deltaTime;
+            }
+        }
+
+        public void StartRising()
+        {
+            _isRising = true;
+        }
+
+        private void resetLava(Vector3 _)
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaRise.cs:38]: `ResetLava`

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaTrigger.cs [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+            {
+                _lava.StartRising();
+            }
+        }
+    }
+}
+
+ 
+
+
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/LavaTrigger.cs:21]: Remove these empty lines

File: Where_Is_My_Wife/Assets/_Main/Scripts/Stage/Lavacanon.cs [hunk 15]
```diff
@@ -0,0 +1,37 @@
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace WhereIsMyWife.Lavacanon
+{
+public class LavaCannon : MonoBehaviour
```
Comment (Horothenic) [inline Where_Is_My_Wife/Assets/_Main/Scripts/Stage/Lavacanon.cs:6]: The class is not aligned, need a Tab to the right, the entire class

**Review by Horothenic — CHANGES_REQUESTED** [review]

**Review by Horothenic — APPROVED** [review]

**Pull request merged**